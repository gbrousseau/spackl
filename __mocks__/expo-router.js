const React = require('react');
const { View, Text } = require('react-native');
console.log('expo-router mock loaded');
module.exports = {
  useRouter: jest.fn(),
  useLocalSearchParams: jest.fn(),
  Redirect: ({ href }) => (
    React.createElement(View, null, React.createElement(Text, null, `Redirected to ${href}`))
  ),
};