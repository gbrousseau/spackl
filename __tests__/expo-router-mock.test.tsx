import React from 'react';
import { render } from '@testing-library/react-native';
import { Redirect } from 'expo-router';
jest.mock('expo-router', () => {
  const React = require('react');
  const { View, Text } = require('react-native');
  return {
    Redirect: ({ href }) => (
      React.createElement(View, null, React.createElement(Text, null, `Redirected to ${href}`))
    ),
  };
});

describe('expo-router manual mock', () => {
  it('should use the mocked Redirect', () => {
    const { toJSON } = render(<Redirect href="/test" />);
    expect(toJSON()).toMatchSnapshot();
  });
});