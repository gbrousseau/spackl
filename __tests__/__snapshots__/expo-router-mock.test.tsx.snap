// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expo-router manual mock should use the mocked Redirect 1`] = `
<View>
  <Text>
    Redirected to /test
  </Text>
</View>
`;
